(DOCSP-24326): Updating template app descriptions (#209)
mongodb/docs-app-services: source/reference/template-apps.txt

You can get up and running with an Atlas App Services App using one of our premade template apps.
Template apps bring together many of the building blocks available in Atlas App Services
and start you off with a prebuilt app that you can customize. You can create 

> an app based on a template using any of the following:

Reviewer: ```suggestion
an app based on a template using any of the following interfaces:
```

what you think of this?

Author: sgtm

         If you want to use the iOS with UIKit template app, select :guilabel:`SwiftUI + Atlas Device Sync Starter`
         in the list of template apps, but specify ``ios.swift.todo`` when you pull the

>          client code using the {+cli-bin+} .

Reviewer: ```suggestion
         client code using the {+cli-bin+}.
```

2. Follow on-screen guides to customize your app.

> 3. Use the {+cli-bin+} to download the code for your app so you can build the front end

Reviewer: ```suggestion
3. Use the {+cli-bin+} to download the code for your app so you can build the frontend
```

i think generally we do as 1 word.

Author: [style guide](https://www.mongodb.com/docs/meta/style-guide/terminology/alphabetical-terms/#f) has it as two (hyphenated if an adjective), which is why I updated it.

> 3. Use the {+cli-bin+} to download the code for your app so you can build the front end

Reviewer: ```suggestion
3. Use the {+cli-bin+} to download the code for the template app so you can build the front end
```

   (if one is included) and interact with the app through the cli.

   If you already have {+cli-bin+} set up, you can use the following {+cli-bin+}
   :ref:`pull command <realm-cli-pull>` to download the code. Otherwise,
   click the :guilabel:`Pull front-end code` button in the welcome

>    section and follow the instructions that display.

Reviewer: >    section and follow the instructions that display.

might be worth specifying *where* the instructions display. ie:

- the instructions that display in the terminal.
- the instructions in the terminal.
- etc.

Author: tweaked styling to sub-bullets to (hopefully) make it more clear that this is referring to instructions that display in the UI

.. tip::

   All of the available template apps except "Manage Database Views" come with a

>    front-end to-do list app.

Reviewer: can we cut this line? 

i don't think this is true anymore b/c of the templates in  https://www.mongodb.com/docs/atlas/app-services/sync/data-access-patterns/flexible-sync-permissions-guide/

also in the future we def might make templates that are backend only.

more maintainable to remove.

Author: I'll defer to you on this, though I figured this was noted because the 'Manage DB Views' template app is the only one that doesn't include any frontend code for you to pull.

   * - Manage Database Views
     - ``triggers``
     - Event-driven :ref:`Database Trigger <database-trigger>` template to update a view in a separate collection. Does *not* include a front-end app.

   * - Flutter + Realm Flutter SDK
     - ``flutter.todo.flex``

>      - Cross-platform to-do list app for Android, iOS, Windows, MacOS, and Linux written in Dart using the :ref:`Realm Flutter SDK <flutter-intro>`.

Reviewer: can you add line abt how flitter using sync as well here? 

i think it's not currently included b/c the flutter template app was local only for a short period of time.

Author: updated

   * - React.JS + Realm Web SDK Boilerplate
     - ``web.mql.todo``
     - Hosted to-do list web app using the :ref:`Realm Web SDK <web-intro>`.
   
   * - GraphQL + React App Boilerplate
     - ``web.graphql.todo``

>      - Hosted to-do list app using the :ref:`GraphQL API <graphql-api>`.

Reviewer: ```suggestion
     - Hosted to-do list web app using the :ref:`GraphQL API <graphql-api>`.
```

   * - Kotlin SDK + Atlas Device Sync Starter (Flexible Sync)
     - ``android.kotlin.todo.flex``

>      - Android to-do list app written in Kotlin using the :ref:`Realm Kotlin SDK <kotlin-intro>` to sync data to App Services's server. Uses :ref:`Flexible Sync <flexible-sync>`.

Reviewer: i noticed that throughout these descriptions it says "to sync data to App Service's server". 

i think this is an odd way of describing what Device Sync does. 

i think it could be better described as something like:

> Android to-to list app written in Kotlin using the Realm Kotlin SDK. Syncs local data to MongoDB Atlas using Device Sync.

this change should be applied to all the template apps that use Device Sync.

Author: updated for all applicable bullets
